File: TransCoder_tokenized_test_set_functions/java/COUNT_EVEN_LENGTH_BINARY_SEQUENCES_WITH_SAME_SUM_OF_FIRST_AND_SECOND_HALF_BITS.java
static int countSeq ( int n, int diff ) {
  if ( Math . abs ( diff ) > n ) return 0;
  if ( n == 1 && diff == 0 ) return 2;
  if ( n == 1 && Math . abs ( diff ) == 1 ) return 1;
  int res = countSeq ( n - 1, diff + 1 ) + 2 * countSeq ( n - 1, diff ) + countSeq ( n - 1, diff - 1 );
  return res;
}

File: TransCoder_tokenized_test_set_functions/python/COUNT_EVEN_LENGTH_BINARY_SEQUENCES_WITH_SAME_SUM_OF_FIRST_AND_SECOND_HALF_BITS.py
def countSeq ( n, diff ) :
    if ( abs ( diff ) > n ) :
        return 0
    if ( n == 1 and diff == 0 ) :
        return 2
    if ( n == 1 and abs ( diff ) == 1 ) :
        return 1
    res = ( countSeq ( n - 1, diff + 1 ) + 2 * countSeq ( n - 1, diff ) + countSeq ( n - 1, diff - 1 ) )
    return res
